modules/transitive/src/closure/TransitiveModel.scala

```diff
@@ -24,16 +25,36 @@ import scala.jdk.CollectionConverters._
 import scala.jdk.OptionConverters._
 import software.amazon.smithy.model.traits.TraitDefinition
 import software.amazon.smithy.utils.ToSmithyBuilder
-import closure.IdRefVisitor
 
 object TransitiveModel {
+
   def compute(
       model: Model,
       entryPoints: List[ShapeId],
       captureTraits: Boolean,
       captureMetadata: Boolean,
       validateModel: Boolean
   ): Model = {
+    val shapes = computeWithVisited(
+      model = model,
+      entryPoints = entryPoints,
+      captureTraits = captureTraits,
+      visitedShapes = mutable.Set.empty
+    )
+    createModelFromShapes(
+      initialModel = model,
+      shapes = shapes,
+      validateModel = validateModel,
+      captureMetadata = captureMetadata
+    )
+  }
+
+  private[closure] def computeWithVisited(
+      model: Model,
+      entryPoints: List[ShapeId],
+      captureTraits: Boolean,
+      visitedShapes: mutable.Set[Shape]
```

Comment:
this is one of the main additions: keeping track of visited shapes

Reply:
This is fine by me